modules/svg/image_loader_svg.cpp

```diff
@@ -134,16 +142,38 @@ Error ImageLoaderSVG::create_image_from_utf8_buffer(Ref<Image> p_image, const ui
 	Vector<uint8_t> image;
 	image.resize(width * height * sizeof(uint32_t));
 
-	for (uint32_t y = 0; y < height; y++) {
-		for (uint32_t x = 0; x < width; x++) {
-			uint32_t n = buffer[y * width + x];
-			const size_t offset = sizeof(uint32_t) * width * y + sizeof(uint32_t) * x;
+	// RGBA to BGRA
+#ifdef __AVX__
+	{
+		uint8_t *dst = image.ptrw();
+		const uint32_t wh = width * height;
+		const __m128i mask = _mm_setr_epi8(2, 1, 0, 3, 6, 5, 4, 7, 10, 9, 8, 11, 14, 13, 12, 15);
+		for (uint32_t i = 0; i < wh / 4; ++i) {
+			__m128i s4 = _mm_loadu_epi8(&buffer[i * 4]);
+			__m128i d4 = _mm_shuffle_epi8(s4, mask);
```

Comment:
More performant intrinsic functions such as `_mm256_shuffle_epi8` or `_mm512_shuffle_epi8` can be used with recent x86/64 processors. In here, I use widely available classic intrinsic function.

I didn't write NEON intrinsics for this. I guess not many ppl run the engine on ARM processors right now.

Reply:
Engine is quite popular on macs with their ARM chips so it might be a good idea to include.

Reply:
@Aleksandar-Argead 
Added NEON code in 335c0dd41cf2a8e74c587849a2f223e431634abd I tested with Raspberry Pi 5.

Reply:
I also added optimized code for AVX2 and AVX512.

x64 SIMD is very fragmented (https://github.com/InstLatx64/InstLatX64_Misc/blob/main/SIMD_Euler/20240124_x64_SIMD_Euler_v114.png) so runtime CPU features checking with CPUID is needed for single executable file to support wide variety of x86-64 processors. I'm not sure if developers want Redot Engine to do that though.